library/src/main/java/com/novoda/downloadmanager/LiteDownloadBatchStatus.java

```diff
@@ -161,9 +161,9 @@ private void updateStatusAsync(Status status, DownloadsBatchStatusPersistence pe
 
     @Nullable
     @Override
-    public DownloadError.Type getDownloadErrorType() {
+    public DownloadError downloadError() {
         if (downloadError.isPresent()) {
-            return downloadError.get().type();
+            return downloadError.get();
```

Comment:
I know is not the goal of this PR but I thought I'd ask (and perhaps amend in other PR) 
Why don't we return here something like:
 `return downloadError.or(new DownloadError(DownloadError.Type.UNKNOWN));` 
to avoid the `null`?

Reply:
We could do but then you can have a `DOWNLOADING` status with an error type `UNKNOWN`, its valid to return null or optional in this case.

Reply:
Ok conversation for another PR, let's get this one merged with the fixes it contains already 👍